.github/workflows/e2e-tests.yml

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright (C) 2015 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: E2E Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  schedule:
+    - cron: '0 1 * * *' # Everyday at 1
+
+env:
+  IT_REVISION: master
+  IT_DIR: kubernetes-itests
+
+jobs:
+  minikube:
+    name: K8S
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        kubernetes: [v1.18.3,v1.12.0]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Minikube-Kubernetes
+        uses: manusa/actions-setup-minikube@v1.0.2
+        with:
+          minikube version: v1.9.2
+          kubernetes version: ${{ matrix.kubernetes }}
+          github token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Setup Java 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: '11'
+      - name: Install Kubernetes Client
+        run: mvn -f pom.xml -B -DskipTests clean install
+      - name: Install and Run Integration Tests
+        run: |
+          cd $IT_DIR
+          mvn test -Dtest="io.fabric8.kubernetes.**"
+  openshift:
+    name: OpenShift
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        openshift: [v3.11.0,v3.10.0]
+    steps:
+      - name: Free up Space
+        run: |
+          sudo apt list --installed
+          sudo apt-get purge -y                     \
+            'aspnetcore.*'                          \
+            'cabal-install.*'                       \
+            'clang.*'                               \
+            'dotnet.*'                              \
+            'firefox.*'                             \
+            'ghc.*'                                 \
+            'google-chrome.*'                       \
+            'heroku.*'                              \
+            'libmagic.*'                            \
+            'libmono.*'                             \
+            'mono-.*'                               \
+            'mysql.*'                               \
+            'php.*'                                 \
+            'ruby.*'
+          sudo apt-get autoremove
+          df -h
```

Comment:
Hopefully this part won't be needed. 
This is only needed in JKube because we run parallel tests with different base images and OpenShift won't start Pods if the 80% free space threshold is surpassed.
This step adds an overhead of ~4 minutes in average to the workflow. If we get Disk pressure in this project too, we should try to use lighter images to perform our tests or other approaches (this should be our last resort).